### PR TITLE
OOP School Library Milestone: Decorate a class

### PR DESCRIPTION
**Pull request by CarlosIgreda**
For this milestone I:

✅ Used GitFlow
✅ Implemented composition (as an example of the design pattern)
✅ Ran a program using the command line

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -23,3 +36,38 @@ def generate_id
...
+    super.capitalize
+  end
+end
+
+# TrimmerDecorator
+class TrimmerDecorator < Decorator
+  def correct_name
+    super[0...10]
+  end
+end
```
Comment (victorokoroji) [inline person.rb]: - [x] I must commend your effort on this project. I think your project will be much better if you follow some best practices, e.g putting all your classes in person.rb file is misleading and can confuse anyone visiting your code.

so I will suggest that you separate each of these classes into there separate files 👍🏽

File: person.rb [hunk 2]
```diff
@@ -23,3 +36,38 @@ def generate_id
     Random.rand(1..500)
   end
 end
+
+# Base Decorator
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```
Comment (victorokoroji) [inline person.rb]: - [x] You did good defining the @nameable variable, kindly note that without an attribute accessor we would not be able to modify this variable outside its file which is not what we want to do. So kindly add it.

e.g 

`    attr_accessor :nameable`

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @CarlosIgreda , 

You have done a great job so far! 💪 👏
You still need to work on a few issues before moving on to the next project, but you're almost there!

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- Decoratiors were properly created ✅ 
- The PR title and description are professional and descriptive ✅ 


## Required Changes ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (CarlosIgreda) [reply person.rb]: @vickymarz, thanks for your feedback. The required changes are done.

File: person.rb (hunk 2 above)
Comment (CarlosIgreda) [reply person.rb]: @vickymarz, thanks for your feedback. The required changes are done.

**Review by jerryowusu — APPROVED** [review]
Hi @CarlosIgreda 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

200w

## Highlights
Used Gitflow :+1: 
Made requested changes :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by CarlosIgreda** [comment]
@jerryowusu , thanks for your feedback. The required changes are done.

**Pull request merged**